examples/demos/swarm_demo/README.md

```diff
@@ -1,17 +1,89 @@
-# Demo application
+# Swarm wih dynamic Trajectories demo application
 
-This folder contains an app layer application for the Crazyflie that is used as a demo.
+This folder contains both an app layer application for all the Crazyflies that are used at the demo and a control_tower python script that supervises the whole process.It also generates the paths for the and generates the trajectories for the Crazyflies.
 
-## Build
+## <u>Building the App</u>
 
-Make sure that you are in the demo folder (not the main folder of the crazyflie firmware). Then type the following to build and flash it while the crazyflie is put into bootloader mode:
+### Crazyflie App Layer-Firmware
+Before buildind the app you need to clean the build space and get the default configuration.
 
 ```
 make clean
+make cf2_defconfig
+```
+
+### 2 Lighthouse base stations
+Make sure that you are in the demo folder (not the main folder of the crazyflie firmware). Then type the following to build the application:
+
+```
 make
+```
+
+### More than 2 Lighthouse base stations
+If you want to run the demo with more than 2 lighthouse base stations, you need to build the application through the menuconfig option running the following command:
+
+```
+make clean
+make cf2_defconfig
+make menuconfig
+```
+Change  the option ```"Expansion Deck Configuration-->Max Number of Base stations"``` to the number you want to and select ```"App Layer COnfiguration-->Enable app entry point"``` while  following the instructions for the [app layer entry-point](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/app_layer/).
+After saving the kbuild configuration, it is also needed to add this line in the Kbuild file ```src/Kbuild```
+```
+obj-y += ../examples/demos/swarm_demo/
+```
+
+Finally,type the following to build the application:
+
+```
+make 
+```
+
+## <u>Flashing</u>
+In order to flash the application to each Crazyflie, you can use the following command:
+
+```
 make cload
 ```
 
-## Running the demo
+or run the ```cload-all.sh``` script which automatically uploads the firmware to all of them . 
+
+
+
+## Control Tower 
+This demo also uses an optimization engine solver for solving the path planning problem. This  [OpEn](https://alphaville.github.io/optimization-engine/) is a  fast solver that is used to generate the paths for all the Crazyflies.In order to install it you need to follow the installation instructions [here](https://alphaville.github.io/optimization-engine/docs/installation) but only the **OpEn Requirements: Rust and clang** and **Python Interface** steps are needed.
+
+Then you will have to go to the following directory  
+
+```
+cd examples/demos/swarm_demo/control_tower/multi_mav_planning/optim_problem
+```
+and run the ```  solver_creation_multiple.py ``` script.
```

Comment:
Getting an FileNotFoundError: [Errno 2] No such file or directory: 'cargo'. Probably the script should check if such a folder exists first and create one if there is none